File: backend/src/main/java/com/example/backend/domain/vote/dto/VoteResponseDto.java
package com.example.backend.domain.vote.dto;

import com.example.backend.domain.vote.entity.Vote;
import lombok.Builder;
import lombok.Getter;

import java.time.LocalDateTime;

@Builder
@Getter
public class VoteResponseDto {
    private Long id;
    private String location;
    private String address;
    private Double latitude;
    private Double longitude;
//    private Integer voterCount;
    private LocalDateTime createdAt;

    public static VoteResponseDto toDto(Vote vote) {
        return VoteResponseDto.builder()
                .id(vote.getId())
                .location(vote.getLocation())
                .address(vote.getAddress())
                .latitude(vote.getLatitude())
                .longitude(vote.getLongitude())
//                .voterCount(vote.getVoterCount())
                .createdAt(vote.getCreatedAt())
                .build();
    }
}

File: backend/src/main/java/com/example/backend/domain/vote/dto/VoteResponseDto.kt
package com.example.backend.domain.vote.dto

import com.example.backend.domain.vote.entity.Vote
import com.fasterxml.jackson.annotation.JsonFormat
import com.fasterxml.jackson.annotation.JsonProperty
import java.time.LocalDateTime

data class VoteResponseDto(
    @JsonProperty("id")
    val id: Long?,
    @JsonProperty("location")
    val location: String,
    @JsonProperty("address")
    val address: String,
    @JsonProperty("latitude")
    val latitude: Double,
    @JsonProperty("longitude")
    val longitude: Double,
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd HH:mm:ss")
    @JsonProperty("createdAt")
    val createdAt: LocalDateTime?
) {
    companion object {
        fun toDto(vote: Vote): VoteResponseDto {
            return VoteResponseDto(
                id = vote.id,
                location = vote.location,
                address = vote.address,
                latitude = vote.latitude,
                longitude = vote.longitude,
                createdAt = vote.createdAt
            )
        }
    }
}